### PR TITLE
Theming support for apps

### PR DESCRIPTION
**Pull request by juliusknorr**
This pr provides the following enhancements for app developers, who want to add support for theming:

CSS classes:
.nc-theming-main-background  Background in theming color
.nc-theming-main-text Text in theming color
.nc-theming-contrast Text in white/black to be shown in front of the theming color

JavaScript Object OCA.Theming
- Access theming variables from javascript

with https://github.com/nextcloud/notifications/pull/9 this fixes notification icon in top left is white instead of black in https://github.com/nextcloud/server/issues/735
- [x] add documentation for app developers to https://github.com/nextcloud/documentation

### CODE REVIEW

File: apps/theming/appinfo/routes.php [hunk 1]
```diff
@@ -55,5 +55,10 @@
 		'url' => '/loginbackground',
 		'verb' => 'GET',
 	],
+	[
+		'name' => 'Theming#getJavascript',
+		'url' => '/js/theming.js',
```
Comment (LukasReschke) [inline apps/theming/appinfo/routes.php]: This did cause problems with some Nginx configs in the past. I'd advociate to keep the `.js` out at the end. Just to be on the safe side. As long as the mimetype is correct this shouldn't harm.

Comment (juliushaertl) [inline apps/theming/appinfo/routes.php]: Just pushed the update. Thanks for the hint.

**Comment by MorrisJobke** [comment]
Code looks good and works 👍

**Comment by schiessle** [comment]
:+1:

**Comment by MorrisJobke** [comment]
> add documentation for app developers to https://github.com/nextcloud/documentation

I opened an issue in the documentation repo.

**Comment by MorrisJobke** [comment]
@juliushaertl Could I ask you to open the backport PR for stable10? Thanks :)

**Comment by juliushaertl** [comment]
@MorrisJobke done.

backport at #833

**Pull request merged**